intermediate-week-1/controllers/auth/login.js

```diff
@@ -1,10 +1,39 @@
+const User = require('../../models/User');
+const Token = require('../../models/Token');
+const errorHandler = require('../../error/errorHandler');
+const { generateAccessToken, generateRefreshToken } = require('../../utils/generateToken');
+
 const login = async (req, res, next) => {
-  /*send a response with the following format if the login is successful
-    *{
-        accessToken: **********,
-        refreshToken: *********
+  try {
+    const { email, password } = req.body;
+
+    const user = await User.findOne({ email }).select('+password');
+
+    if (user && (await user.matchPassword(password))) {
+      const accessToken = generateAccessToken(user._id, user.firstName, user.lastName, user.email);
+      const refreshToken = generateRefreshToken(user._id, user.firstName, user.lastName, user.email);
+      // user.token = refreshToken;
+      const token = Token({
+        token: refreshToken,
+        user: user._id,
+      });
+      await token.save();
+      return res.json({
+        accessToken,
+        refreshToken,
+      });
+
+      // next();
+    } else {
+      return res.sendApiError({
+        status: 404,
+        title: 'Invalid data!',
+        detail: 'Invalid email or password',
+      });
     }
-    */
+  } catch (error) {
```

Comment:
Nice error handling, but this isn;t exactly how the errorhandler was meant to be used. 

Kindly watch the videos in this playlist. 

https://www.youtube.com/playlist?list=PLx6gGOAOEqGHqDaIjFhwnojD4Lhf4kU_N

From 
111 handling unhandled routes  down to 
122 Catching Uncaught Exceptions. 

Well done once again